### PR TITLE
feat: add BarCodeDetector

### PR DESCRIPTION
**Pull request by charlesmindee**
This PR adds a BarCodeDetector implemented with Open CV in the artefacts sub-module. 
For now, only horizontal or small-rotated bar_codes are supported. 
Samples are displayed below, any feedback is welcome!
bar1
bar2
bar3
bar4
bar5
bar6
bar7

### CODE REVIEW

**Comment by charlesmindee** [comment]
It takes 0.4s to run this detector 100 times on the same image, a 1024 x 1024 picture with a bar-code on it. Experiments were conducted on a CPU (intel core i7 9th gen).

File: doctr/models/artefacts/barcode.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+# See LICENSE or go to <https://www.apache.org/licenses/LICENSE-2.0.txt> for full license details.
+
+import cv2
+import numpy as np
+from typing import List, Tuple
+
+__all__ = ['BarCodeDetector']
+
+
+class BarCodeDetector:
```
Comment (fg-mindee) [inline doctr/models/artefacts/barcode.py:13]: For now, either we:
- convert this to a function
- add constants in the constructor (min size of barcode, canny edge args, etc)

**Review by fg-mindee — COMMENTED** [review]
Thanks for the PR! I added one comment

File: test/test_artefacts.py [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
+
+
+def test_qr_code_detector(mock_image_folder):
+    detector = BarCodeDetector()
+    for img in os.listdir(mock_image_folder):
+        image = DocumentFile.from_images(os.path.join(mock_image_folder, img))[0]
+        barcode = detector(image)
+        assert len(barcode) == 0
 from doctr.models.artefacts import FaceDetector
 from doctr.documents import DocumentFile
```
Comment (fg-mindee) [inline test/test_artefacts.py:13]: there is an issue here since the merge. Double imports lack of blank lines

**Review by fg-mindee — CHANGES_REQUESTED** [review]
Flake8 to be fixed in the tests, the rest looks good

**Review by fg-mindee — APPROVED** [review]
Thanks!

**Pull request merged**